master/pkg/searcher/custom_search.go

```diff
@@ -0,0 +1,138 @@
+package searcher
+
+import (
+	"encoding/json"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
+	"github.com/determined-ai/determined/proto/pkg/experimentv1"
+)
+
+type (
+	customSearchState struct {
+		// store the operations
+		// store the events
+		SearchMethodType   SearchMethodType `json:"search_method_type"`
+		SearcherEventQueue *SearcherEventQueue
+	}
+
+	customSearch struct {
+		defaultSearchMethod
+		expconf.CustomConfig
+		customSearchState
+		eventCount int32
+	}
+)
+
+func newCustomSearch(config expconf.CustomConfig) SearchMethod {
+	return &customSearch{
+		CustomConfig: config,
+		customSearchState: customSearchState{
+			SearchMethodType:   CustomSearch,
+			SearcherEventQueue: createSearcherEventQueue(),
+		},
+	}
+}
+
+func createSearcherEventQueue() *SearcherEventQueue {
+	return newSearcherEventQueue()
+}
+func (s *customSearch) initialOperations(ctx context) ([]Operation, error) {
+	s.eventCount++
+	event := experimentv1.SearcherEvent_InitialOperations{
+		InitialOperations: &experimentv1.InitialOperations{},
+	}
+	searcherEvent := experimentv1.SearcherEvent{
+		Id:    s.eventCount,
+		Event: &event,
+	}
+
+	err := s.SearcherEventQueue.Enqueue(&searcherEvent)
+	return nil, err
+}
+
+func (s *customSearch) GetSearcherEventQueue(context) *SearcherEventQueue {
+	return s.SearcherEventQueue
+}
+
+func (s *customSearch) trialCreated(context, model.RequestID) ([]Operation, error) {
+	s.eventCount++
+	event := experimentv1.SearcherEvent_TrialCreated{
+		TrialCreated: &experimentv1.TrialCreated{},
```

Comment:
The request IDs and other fields should be filled in for all of these events.

Reply:
I'm going to do this as part of the next ticket.